### PR TITLE
Support revised documents/collections in Jekyll 3

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
Changes to this plugin to ensure it works with the latest release of Jekyll 3.

(Not sure if these changes will break compatibility with Jekyll 2; is that something that still needs to be figured into the plugin @alfredxing?)

### CODE REVIEW

**Comment by alfredxing** [comment]
Thanks for the PR!

Ideally, we would continue to support Jekyll 2. Maybe we can do a version check of some sort? Or just check for `@posts.docs`?

@parkr Any thoughts?

**Comment by parkr** [comment]
> @parkr Any thoughts?

You could do a version check, absolutely!

**Comment by paulrobertlloyd** [comment]
@parkr/@alfredxing I’m testing for `@posts.docs` rather than for a specific version of Jekyll. Not sure if that’s the correct approach, but seemed more fool proof (and I wasn’t sure how to reliably test for a Jekyll version). I’ve tested this locally, so should work! Finger crossed…

File: lib/jekyll-archives.rb [hunk 1]
```diff
@@ -131,7 +131,13 @@ def post_attr_hash(post_attr)
         # Build a hash map based on the specified post attribute ( post attr =>
         # array of posts ) then sort each array in reverse order.
         hash = Hash.new { |h, key| h[key] = [] }
-        @posts.each { |p| p.send(post_attr.to_sym).each { |t| hash[t] << p } }
+
+        # In Jekyll 3, Collection#each should be called on the #docs array directly.
```
Comment (paulrobertlloyd) [inline lib/jekyll-archives.rb]: @parkr I’ve copied the deprecation message here, I hope this is the correct description of what’s happening.

File: lib/jekyll-archives.rb [hunk 2]
```diff
@@ -147,7 +153,13 @@ def categories
       # Custom `post_attr_hash` method for years
       def years
         hash = Hash.new { |h, key| h[key] = [] }
-        @posts.each { |p| hash[p.date.strftime("%Y")] << p }
+
+        # In Jekyll 3, Collection#each should be called on the #docs array directly.
```
Comment (paulrobertlloyd) [inline lib/jekyll-archives.rb]: @parkr I’ve used the same deprecation message here as above; not sure if it needs to be slightly different?

**Comment by alfredxing** [comment]
Thanks! Looks good to me, maybe @parkr can get one last set of :eyes: on it when he gets the chance?

**Comment by parkr** [comment]
@alfredxing Why is `post_attr_hash` overridden?

**Comment by alfredxing** [comment]
To be honest I'm not quite sure... do you think that just calling `@site.post_attr_hash` would work?

**Comment by parkr** [comment]
@alfredxing try it! It should work just fine. If your tests pass, then all is well?

**Comment by alfredxing** [comment]
Closing in favour of #53.

**Pull request closed without merging**